cli/azd/pkg/tools/azcli/ad.go

```diff
@@ -55,8 +56,10 @@ func (cli *azCli) CreateOrUpdateServicePrincipal(
 		return nil, fmt.Errorf("failed resetting application credentials: %w", err)
 	}
 
+	// Required role assignment for applying role assignment
+	checkRoles := []string{"Owner", "User Access Administrator"}
 	// Apply specified role assignment
-	err = cli.ensureRoleAssignments(ctx, subscriptionId, roleName, servicePrincipal)
+	err = cli.ensureRoleAssignments(ctx, subscriptionId, roleName, servicePrincipal, checkRoles)
```

Comment:
`ensureRoleAssignments` is too late to check if the user has the required roles.
This method is about adding the roles to the service principal after it is created.   We don't want to create the service principal if we know we can't assign the roles.

The roles validation must be performed before calling `CreateOrUpdateServicePrincipal` , as early as possible and before doing any other operation